crates/chain/tests/test_keychain_txout_index.rs

```diff
@@ -367,3 +368,49 @@ fn test_non_wildcard_derivations() {
     assert_eq!(revealed_spks.count(), 0);
     assert!(revealed_additions.is_empty());
 }
+
+#[test]
+#[allow(clippy::iter_nth_zero)]
+fn test_spkiterator_wildcard() {
+    let (_, external_desc, _) = init_txout_index();
+    let external_spk_0 = external_desc.at_derivation_index(0).script_pubkey();
+    let external_spk_16 = external_desc.at_derivation_index(16).script_pubkey();
+    let external_spk_20 = external_desc.at_derivation_index(20).script_pubkey();
+    let external_spk_21 = external_desc.at_derivation_index(21).script_pubkey();
+    let external_spk_max = external_desc
+        .at_derivation_index(BIP32_MAX_INDEX)
+        .script_pubkey();
+
+    let mut external_spk = SpkIterator::new(Cow::Owned(external_desc));
+    let max_index = BIP32_MAX_INDEX - 22;
+
+    assert_eq!(external_spk.next().unwrap(), (0, external_spk_0));
+    assert_eq!(external_spk.nth(15).unwrap(), (16, external_spk_16));
+    assert_eq!(external_spk.nth(3).unwrap(), (20, external_spk_20));
+    assert_eq!(external_spk.next().unwrap(), (21, external_spk_21));
+    assert_eq!(
+        external_spk.nth(max_index as usize).unwrap(),
+        (BIP32_MAX_INDEX, external_spk_max)
+    );
+    assert_eq!(external_spk.nth(0), None);
```

Comment:
It'll be good to have tests for checking the `end` bound is enforced with a wildcard descriptor.